src/agent_workflow_server/agents/adapters/llamaindex.py

```diff
@@ -25,16 +30,42 @@ def load_agent(self, agent: object) -> Optional[BaseAgent]:
 class LlamaIndexAgent(BaseAgent):
     def __init__(self, agent: Workflow):
         self.agent = agent
+        self.contexts: Dict[str, Dict] = {}
 
     async def astream(self, run: Run):
         input = run["input"]
-        handler = self.agent.run(**input)
+        ctx_data = self.contexts.get(run["thread_id"])
+
+        handler = self.agent.run(
+            ctx=Context.from_dict(self.agent, ctx_data) if ctx_data else None,
+            input={**input},
+        )
+        if handler.ctx is None:
+            # This should never happen, workflow.run actually sets the Context
+            raise ValueError("Context cannot be None.")
+
+        if "interrupt" in run and "user_data" in run["interrupt"]:
+            user_data = run["interrupt"]["user_data"]
+
+            # FIXME: workaround to extract the user response from a dict/obj. Needed for input validation, remove once not needed anymore.
+            if isinstance(user_data, dict):
+                user_data = list(user_data.values())[0]
```

Comment:
if user_data={}
it is an instance of dict, but list(user_data.values())[0] will trigger an IndexError exception